eth/db/storage.py

```diff
@@ -120,7 +141,7 @@ def __delitem__(self, key: bytes) -> None:
 
     @property
     def has_changed_root(self) -> bool:
-        return self._write_trie and self._write_trie.root_hash != self._starting_root_hash
+        return self._write_trie
```

Comment:
Assuming I'm correct that this is checking for the `None` value, I'd advocate for this to be `return self._write_trie is not None`.